cairo/tests/utils/args_gen.py

```diff
@@ -496,19 +511,39 @@ def _gen_arg(
             arg_type = Mapping[get_args(arg_type)[0], bool]
 
         data = {
-            _gen_arg(dict_manager, segments, get_args(arg_type)[0], k): _gen_arg(
-                dict_manager, segments, get_args(arg_type)[1], v
-            )
+            _gen_arg(
+                dict_manager,
+                segments,
+                get_args(arg_type)[0],
+                k,
+                hash_mode=hash_mode is not False,
```

Comment:
```suggestion
                hash_mode=hash_mode,
```
`x is not False` <=> `x is in [True or None]`
Suppose x = None, `x is not False` will be True
I do not think we want to hash in the default case when hash_mode == None

If not mistaken you can pass directly hash_mode
The condition below is 
```
if hash_mode:
      return tuple(felt_values)
```
so will enter it only if hash_mode == True